docs/decisions/2025-03-31-streaming-responses-for-chatbot.md

```diff
@@ -0,0 +1,166 @@
+# Implementing Streaming Responses for Chatbot
+
+- Status: proposed
+- Deciders: Kevin Boyer, Yoom Lam
+- Date: 2025-03-31
+
+## Context and Problem Statement
+
+Our chatbot currently uses a request-response model, where users submit a query and wait for the entire response (often >10 seconds) before seeing any output. This approach has some limitations:
+
+- **Delayed feedback:** Users experience uncertainty and disengagement while waiting for the full response.
+- **Connection vulnerability:** Longer-running requests are at risk of being dropped due to intermittent connectivity, causing users to lose responses entirely.
+
+Implementing streaming responses addresses these issues by providing immediate visual feedback as content is generated, improving perceived responsiveness, and reducing the risk of losing responses due to connectivity issues.
+
+## Decision Drivers
+
+* User experience (responsiveness, feedback, and engagement)
+* Reliability of connection to the chatbot
+* Implementation effort for both Nava and third-party clients
+* Maintainability
+
+## Solution Options
+
+### Polling
+
+- Handles intermittent connectivity through client-side retry logic
+- Introduces latency dependent on polling intervals  
+- Requires writing client-side logic for managing accumulated response chunks and state.
+
+### Server-Sent Events (SSE)
+
+- Immediate/event-driven updates provide a responsive user experience
+- Built-in browser support simplifies client-side implementation 
```

Comment:
Browser support is less for SEE than websockets. https://www.lambdatest.com/web-technologies/eventsource
> Server-sent events on IE is not supported on any IE versions. Server-sent events on Edge is fully supported on 79-133, and not supported on 12-18 Edge versions. Server-sent events on Firefox is fully supported on 6-138, and not supported on 2-5 Firefox versions.

Reply:
https://sii.pl/blog/en/server-side-events-implementation-and-highlights/:
> All major web browsers, including Chrome, Firefox, Safari, Edge, and Opera, support SSE. IE is the only major browser that doesn’t support SSE.